frontend/src/components/DronePage/DronePage.tsx

```diff
@@ -1,22 +1,40 @@
-import React, { FC } from 'react';
-import { Container, Row } from 'react-bootstrap';
-import { drones } from '../../sample_data';
+import React, { FC, useState } from 'react';
+import { Container, Row, Button } from 'react-bootstrap';
+import { dronesData } from '../../sample_data';
 import { Drone } from '../../../database/schemas/drone_schema';
 import DroneCard from '../DroneCard/DroneCard';
 import './DronePage.css';
 
 interface DronePageProps {}
 
-const DronePage: FC<DronePageProps> = (props) => {
-  const rows = drones.map((drone : Drone) => <Row key={drone.id}><DroneCard drone={drone}></DroneCard></Row>)
+const DronePage: FC<DronePageProps> = (props : DronePageProps) => {
+    const convertRows = (drones : Array<Drone>) => {
+        return drones.map((drone : Drone) => <Row key={drone.id}><DroneCard drone={drone}></DroneCard></Row>);
+    }
+    let initialRows = convertRows(dronesData);
+    const [rows, setRows] = useState(initialRows);
+    
+    let drones = dronesData;
```

Comment:
Right now you are modifying a variable defined in a different file. In react, my understanding is that the recommended practice is to have every variable be immutable, as it can lead to very hard to track down bugs. You can do this with useState or useReducer, but you should never modify state with something like drones.push(). Rather, to push you have to recreate the entire array with a new element (so it's O(n)). Anyways if it works it's fine for this project

Also this let can be const